src/main/scala/com/monsanto/arch/cloudformation/model/simple/Builders.scala

```diff
@@ -244,6 +181,18 @@ trait Subnet extends AvailabilityZone with Outputs {
 
     f(sub) ++ sub.andOutput(name, name)
   }
+
+  def nat(routeTables: Seq[`AWS::EC2::RouteTable`])(implicit s: `AWS::EC2::Subnet`) = {
```

Comment:
This needs to accept the VPC Gateway Attachment resource and set it as a DependsOn for the EIP.  See http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-eip.html